docs/the-new-architecture/pillars-turbomodule.md

```diff
@@ -3,16 +3,699 @@ id: pillars-turbomodules
 title: TurboModules
 ---
 
-This section contains a high-level introduction to TurboModules. It provides enough context to understand when a TurboModule is needed and how it roughly works.
+import Tabs from '@theme/Tabs'; import TabItem from '@theme/TabItem'; import constants from '@site/core/TabsConstants';
 
-This section must have a warning that it works only with the new architecture enabled. It points to the [migration section](../new-architecture-intro).
+If you've worked with React Native, you may be familiar with the concept of Native Modules, which allow JavaScript and platform-native code to communicate over the React Native "bridge", which handles cross-platform serialization via JSON.
 
-## How to create a Turbomodule
+TurboModules are the next iteration on Native Modules that provide a few extra [benefits](./why):
 
-This section is a step-by-step guide to create a TurboModule from scratch. The list of subsections is roughly:
+- Strongly typed interfaces that are consistent across platforms
+- The ability to create shared C++ code for use across platforms
+- Lazy loading of modules, allowing for faster app startup
+- The use of JSI, a JavaScript interface for native code, which allows for more efficient communication between native and JavaScript code than the bridge
 
-- JS spec (with all the supported features)
-- Configuration (package.json, cocoapods, gradle, …) and CodeGen
-- Native code (one section for iOS and one for Android)
-- Integration in an App (`yarn add` and how to connect the JS specs to the app itself)
-- Troubleshooting (common issues and how to solve them)
+This guide will show you how to create a basic TurboModule.
+
+:::caution
+TurboModules only work with the **New Architecture** enabled.
+To migrate to the **New Architecture**, follow the [Migration guide](../new-architecture-intro)
+:::
+
+## How to Create a TurboModule
+
+To create a TurboModule, we need to:
+
+1. Define a set of JavaScript specifications.
+2. Configure the module and inspect the code created by Codegen.
```

Comment:
```suggestion
2. Configure the module and let the Codegen generate the scaffolding.
```

scaffolding can be replaced with boilerplate or other similar terms.

I would just not mention 'inspect the code' in the list of necessary steps.

Reply:
I like scaffolding, I think that's very clear about the role it plays, and implies you're expected to fill in more code.

I'm guessing the original intent in the "inspect the code" wording (from the Fabric Guide) was to talk about how we're going to read the scaffolding generated so we understand what native code to write, which does seem like a useful part of the process to convey, to me?

Reply:
Also just a note, but a few different suggestions left mention using "the CodeGen". Just to make sure we're clear, I don't think this is grammatically correct, it should either be "let CodeGen generate" or "let the CodeGen CLI tool (or process, or script, etc) generate..."

Reply:
Addressed, but leaving open for discussion if needed

Reply:
> I'm guessing the original intent in the "inspect the code" wording (from the Fabric Guide) was to talk about how we're going to read the scaffolding generated so we understand what native code to write, which does seem like a useful part of the process to convey, to me?

That's correct, that was my intent there. The user will need to read the code generated by CodeGen in order to know how to use the generated types and files.

In this bullet list, I think that both the sentences fit. I will use the one that requires less effort from the reader (I think it is Nico's suggestion)